tools/src/bin/benchmark-clvm-cost.rs

```diff
@@ -264,7 +264,69 @@ fn base_call_time_no_nest(a: &mut Allocator, op: &Operator, per_arg_time: f64) -
         num_samples += 1;
     }
 
-    ((total_time - per_arg_time * num_samples as f64) / num_samples as f64).max(100.0)
+    (total_time - per_arg_time * num_samples as f64) / num_samples as f64
+}
+
+// this adds 32 bytes at a time compared to per_byte which adds 5 at a time
+// at the moment this func is only for sha256tree and will need adapting in the future for other uses
+fn time_per_byte_for_atom(a: &mut Allocator, output: &mut dyn Write) -> (f64, f64) {
+    let mut samples = Vec::<(f64, f64)>::new();
+    let dialect = ChiaDialect::new(ENABLE_SHA256_TREE); // enable shatree
+
+    let op_code = a.new_number(63.into()).unwrap();
+    let quote = a.new_number(1.into()).unwrap();
+    let mut atom_str = "ff".repeat(10_000);
+    let checkpoint = a.checkpoint();
+
+    for i in 0..10000 {
+        // make the atom longer as a function of i
+        atom_str.push_str(&((i % 89) + 10).to_string().repeat(32)); // just to mix it up
+        let atom = a.new_atom(&hex::decode(&atom_str).unwrap()).unwrap();
+        // let args = a.new_pair(atom, a.nil()).unwrap();
+        let args = a.new_pair(quote, atom).unwrap();
+        let call = a.new_pair(args, a.nil()).unwrap();
+        let call = a.new_pair(op_code, call).unwrap();
+        let start = Instant::now();
+        run_program(a, &dialect, call, a.nil(), 11000000000).unwrap();
+        let duration = start.elapsed();
+        let sample = (i as f64, duration.as_nanos() as f64);
+        writeln!(output, "{}\t{}", sample.0, sample.1).expect("failed to write");
+        samples.push(sample);
+
+        a.restore_checkpoint(&checkpoint);
+    }
+
+    linear_regression_of(&samples).expect("linreg failed")
+}
+
+// at the moment this func is only for sha256tree and will need adapting in the future for other uses
+fn time_per_cons_for_list(a: &mut Allocator, output: &mut dyn Write) -> (f64, f64) {
+    let mut samples = Vec::<(f64, f64)>::new();
+    let dialect = ChiaDialect::new(ENABLE_SHA256_TREE); // enable shatree
+
+    let op_code = a.new_number(63.into()).unwrap();
+    let quote = a.new_number(1.into()).unwrap();
+    let mut list = a.nil();
+
+    for _ in 0..500 {
+        list = a.new_pair(a.nil(), list).unwrap();
+    }
+
+    for i in 0..10000 {
```

Comment:
I think you could probably add `.step(500)` here, and add 500 entries in each step. i.e. move the above loop into here